lib/dependencies.sh

```diff
@@ -87,14 +87,34 @@ log_build_scripts() {
 
 yarn_node_modules() {
   local build_dir=${1:-}
+  local production=${YARN_PRODUCTION:-false}
 
   echo "Installing node modules (yarn.lock)"
   cd "$build_dir"
-  yarn install --frozen-lockfile --ignore-engines 2>&1
+  yarn install --production=$production --frozen-lockfile --ignore-engines 2>&1
+}
+
+yarn_prune_devdependencies() {
+  local build_dir=${1:-} 
+
+  if [ $NODE_ENV == "test" ]; then
+    echo "Skipping because NODE_ENV is 'test'"
+    return 0
+  elif [ $NODE_ENV != "production" ]; then
```

Comment:
It might be worth quoting the `$NODE_ENV` and the various other variables in conditionals, since otherwise spurious spaces can cause a `too many arguments` bash error and the wrong result.

eg:
```bash
$ NODE_ENV="foo bar"; if [ $NODE_ENV != "production" ]; then echo "not production"; else echo "production"; fi
bash: [: too many arguments
production
```

Fwiw [shellcheck](https://github.com/koalaman/shellcheck) is able to find issues like this (the Python buildpack runs it on Travis):
```bash
In lib/dependencies.sh line 103:
  elif [ $NODE_ENV != "production" ]; then
         ^-- SC2086: Double quote to prevent globbing and word splitting.
```
(https://github.com/koalaman/shellcheck/wiki/SC2086)

Reply:
*cough* shellcheck is the best